### PR TITLE
rel: Prepare v4.3.0 release

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?

Prepares the next release of libhoney-js.

## Short description of the changes

- Update version to 4.3.0
- Add changelog entry
- Update README with updated Node version (14.18)
- Update min node version in package.json

### CODE REVIEW

**Review by JamieDanielson — COMMENTED** [review]
What is the minimum node version bump from? Is that worth marking as a breaking change?

**Comment by MikeGoldsmith** [comment]
It's from the superagent update. We already had 14+ as a min version, this updates to 14.18. 

Yeah, we can add a breaking change note to the change log.

File: README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ For tracing support and automatic instrumentation of Express and other common li
 
 ## Dependencies
 
-**Node 14+**
+**Node 14.18+**
```
[the remark this answers is not in the document]
Comment (MikeGoldsmith) [reply README.md:16]: Ah, nice. Yes, we can add to package.json

[the remark this answers is not in the document]
Comment (MikeGoldsmith) [reply README.md:16]: Added in https://github.com/honeycombio/libhoney-js/pull/416/commits/6608e9343200d8834764223d036bb3b292b50717

**Review by codeboten — APPROVED** [review]

**Pull request merged**